analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -1383,4 +1384,93 @@ public void Invocation_NullableGetHasValue_UntrackedSymbol()
             x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull),  // Because it's int
             x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, NumberConstraint.From(0)));
     }
+
+    [TestMethod]
+    public void Invocation_EnumerableCount_SetsNumberConstraint()
+    {
+        const string code = """
+                var x = list.Count();
+                Tag("Before", x);
+
+                x = list.Count(x => true);
+                Tag("Predicate", x);
+
+                list.Clear();
+                x = list.Count();
+                Tag("Clear", x);
+
+                list.Add(42);
+                x = list.Count();
+                Tag("Add", x);
+            """;
+        var validator = SETestContext.CreateCS(code, "List<int> list").Validator;
+        validator.ValidateContainsOperation(OperationKind.Invocation);
+
+        validator.TagValue("Before").Should().HaveOnlyConstraints(NumberConstraint.From(0, null), ObjectConstraint.NotNull);
+        validator.TagValue("Predicate").Should().HaveOnlyConstraints(NumberConstraint.From(0, null), ObjectConstraint.NotNull);
+        validator.TagValue("Clear").Should().HaveOnlyConstraints(NumberConstraint.From(0), ObjectConstraint.NotNull);
+        validator.TagValue("Add").Should().HaveOnlyConstraints(NumberConstraint.From(1, null), ObjectConstraint.NotNull);
+    }
+
+    [TestMethod]
+    public void Invocation_CollectionMethods_SetCollectionConstraint()
```

Comment:
Check the second part of invocation, that learns CollectionConstraint based on Add/Remove/Clear methods.